src/client.rs

```diff
@@ -0,0 +1,260 @@
+
+        // Create the query specifications (time, location, etc.)
+        let query_specs = build_query_specs(
+            &start_date, &end_date, &interval, &parameters, &coords_str, &optionals
+        ).await;
+
+        // Get the query result
+        let result = self.do_http_get(&query_specs).await;
+
+        // Match the result
+        // TODO: Check this match statement, when is ApiError produced?
+        match result {
+            Ok(response) => match response.status() {
+                StatusCode::OK => {
+                    let dataframe: polars::frame::DataFrame = parse_response_to_dataframe(response)
+                        .await?;
+        
+                    Ok(dataframe)
+                }
+                status => Err(ConnectorError::HttpError(
+                    status.to_string(),
+                    response.text().await.unwrap(),
+                    status,
+                )),
+            },
+            Err(connector_error) => Err(ConnectorError::ApiError {
+                source: connector_error,
+            }),
+        }
+    }
+    
+    async fn do_http_get(&self, url_fragment: &str) -> Result<Response, reqwest::Error> {
+        let full_url = build_url(url_fragment)
+            .await
+            .expect("URL fragment must be valid");
+
+        println!(">>>>>>>>>> full_url: {}", full_url);
+
+        self.http_client
+            .get(full_url)
+            .basic_auth(&self.username, Some(String::from(&self.password)))
+            .send()
+            .await
+    }
+}
+
+async fn parse_response_to_dataframe(
+    response: Response,
+) -> Result<polars::frame::DataFrame,  polars::error::PolarsError> {
+    // Get the response text:
+    let body = response.text().await.unwrap();
+
+    // Parse the response to a DataFrame
+    let file = std::io::Cursor::new(&body);
+    use polars::prelude::*; 
+    let df = polars::io::csv::CsvReader::new(file)
+        .infer_schema(Some(100))
+        .with_delimiter(b';')
+        .has_header(true)
+        .with_parse_dates(false)
+        .with_ignore_parser_errors(false)
+        .finish();
+    df 
+}
+
+async fn build_query_specs(
+    start_date: &chrono::DateTime<chrono::Utc>,
+    end_date: &chrono::DateTime<chrono::Utc>,
+    interval: &chrono::Duration,
+    parameters: &Vec<String>,
+    coords_str: &str,
+    optionals: &Option<Vec<String>>
+) -> String {
+
+    let query_specs = format!(
+        "{}--{}:{}/{}/{}/csv",
+        start_date.to_rfc3339(),
+        end_date.to_rfc3339(),
+        interval.to_string(),
+        parameters.join(","),
+        coords_str,
+    );
+
+    // Handles optional parameters 
+    let query_specs = match optionals {
+        None => query_specs,
+        Some(_) => {
+            format!(
+                "{}?{}",
+                query_specs,
+                optionals.as_ref().unwrap().join("&")
+            )
+        }
+    };
+    
+    return query_specs
```

Comment:
`return query_specs;` with semicolon at the end or just
`query_specs`
I prefer the 2nd approach, the rust way.